.github/workflows/phpstan.yml

```diff
@@ -0,0 +1,66 @@
+name: Run tests
+
+on:
+  workflow_call:
+    inputs:
+      deps:
+        type: string
+        required: false
+        default: stable
+      continue_on_error:
+        type: boolean
+        required: false
+        default: false
+
+jobs:
+  test:
+    name: PHPStan ${{ matrix.php }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ['7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1']
```

Comment:
is it really needed to run on each PHP version? or maybe enough if run for lowest / highest only?

Reply:
This is good question. Personally I think it's required to run it on every PHP version that is supported because it will detect:
- usage of PHP features not available in specific versions. Of course, this can be also detected by unit tests, but we can't be sure whole codebase is covered with tests, while SA will look at every file.
- inconsistency between vendor packages' signatures: this is most problematic IMHO, for each PHP version set of dependencies is different because of wide constraints. And SA should detect if this code is compatible with every vendor version in supported range... But it is not that easy, [look at this thread](https://mobile.twitter.com/_Codito_/status/1553753386985705484) 🤔

As far as I remember this approach is used in several projects, I can look for examples if you want 😉 Alternative is to write in 8.1 and then use Rector for downgrading to 7.0 (like PHPStan and Rector do), but it is A LOT of work.

Reply:
But @ondrejmirtes [says](#discussion_r935470197) it's not required 😉 Unfortunately, PHP 7.1 can be enforced by config and this library supports 7.0 🤔